pkg/tsdb/cloudmonitoring/time_series_query.go

```diff
@@ -35,6 +35,44 @@ func (timeSeriesQuery cloudMonitoringTimeSeriesQuery) appendGraphPeriod(req *bac
 	return ""
 }
 
+func doRequestQueryPage(query string, nextPageToken string, s *Service, ctx context.Context, dsInfo datasourceInfo, path string, tracer tracing.Tracer, req *backend.QueryDataRequest) (cloudMonitoringResponse, error) {
+	requestBody := map[string]interface{}{
+		"query": query,
+	}
+	if nextPageToken != "" {
+		requestBody["pageToken"] = nextPageToken
+	}
+
+	buf, err := json.Marshal(requestBody)
+	if err != nil {
+		return cloudMonitoringResponse{}, err
+	}
+	r, err := s.createRequest(ctx, &dsInfo, path, bytes.NewBuffer(buf))
+	if err != nil {
+		return cloudMonitoringResponse{}, err
+	}
```

Comment:
you could evaluate creating the request outside of this function and simply modify the `Body` for each page:

```go
	req.Body = io.NopCloser(bytes.NewBuffer(buf))
```

Also (nit), since you don't need to do the nextPageToken management for the first query, you can do that within the `for` loop, not here.

That may save you a bunch of parameters and complexity here but it's just moving things around for style so up to you what you prefer :)